### PR TITLE
fix: Fail order status transition to pending if incomplete attendee info

### LINKED ISSUE #7069: Order should fail to transition to pending state if attendee info is incomplete
Right now, since there was no validation on attendee form, an attendee could submit dummy information and change order status to pending and complete the order. After #7068, order validation should be put into place so that no order can be completed without correct attendee info state

### PR DESCRIPTION
**Pull request by Haider8**
Fixes #7069 

#### Short description of what this resolves:
Right now order status can be transitioned to `pending` even if corresponding attendees have incomplete info, which is incorrect.

#### Changes proposed in this pull request:
Fail order status transitioning to `pending` if incomplete attendee info 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/orders.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712&bid=18997949)

File: app/api/orders.py [hunk 1]
```diff
@@ -379,6 +381,13 @@ def before_update_object(self, order, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
+        if data.get('status') == 'pending':
```
Comment (iamareebjamal) [inline app/api/orders.py:384]: Can be completed or placed as well

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- tests/all/integration/api/helpers/order/test_edit_order.py  2
- app/api/orders.py  2
         

Clones added
============
- tests/all/integration/api/helpers/order/test_edit_order.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712&bid=18997949)

File: tests/all/integration/api/helpers/order/test_edit_order.py [hunk 2]
```diff
@@ -91,3 +92,150 @@ def test_ignore_on_order_event_update(client, db, user, jwt):
...
+                'source': {'pointer': '/data/attributes'},
+                'title': 'Unprocessable Entity',
+                'detail': "Missing required fields ['job_title', 'tax_business_info']",
+            }
+        ],
+        'jsonapi': {'version': '1.0'},
+    }
+
+
+def test_order_status_to_pending_incomplete_complex_custom_form(client, db, user, jwt):
```
Comment (niranjan94) [inline tests/all/integration/api/helpers/order/test_edit_order.py:205]: Codacy found an issue: [Function name "test_order_status_to_pending_incomplete_complex_custom_form" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712)

File: tests/all/integration/api/helpers/order/test_edit_order.py [hunk 3]
```diff
@@ -91,3 +92,150 @@ def test_ignore_on_order_event_update(client, db, user, jwt):
...
+        is_included=True,
+        is_required=False,
+    )
+
+    db.session.commit()
+
+    return str(order.id)
+
+
+def test_order_status_to_pending_incomplete_simple_custom_form(client, db, user, jwt):
```
Comment (niranjan94) [inline tests/all/integration/api/helpers/order/test_edit_order.py:166]: Codacy found an issue: [Function name "test_order_status_to_pending_incomplete_simple_custom_form" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712)

File: app/api/orders.py (hunk 1 above)
Comment (Haider8) [reply app/api/orders.py:384]: So should I just make this `data.get('status') != 'initializing'` ?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/all/integration/api/helpers/order/test_edit_order.py  2
- app/api/orders.py  2
         

Clones added
============
- tests/all/integration/api/helpers/order/test_edit_order.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712&bid=18997949)

File: app/api/orders.py (hunk 1 above)
Comment (iamareebjamal) [reply app/api/orders.py:384]: No, because this should not run order is being cancelled.

**Comment by iamareebjamal** [comment]
Please test manually that placed orders and order cancellation is working

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/all/integration/api/helpers/order/test_edit_order.py  2
- app/api/orders.py  2
         

Clones added
============
- tests/all/integration/api/helpers/order/test_edit_order.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712&bid=18997949)

**Comment by iamareebjamal** [comment]
You have added tests for failure case, great. Also add tests for success case, where attendees have correct data and order is successfully changed to pending state. Just adding for complex custom forms is enough

**Comment by Haider8** [comment]
> Also add tests for success case, where attendees have correct data and order is successfully changed to pending state

Will do this once I cover `placed` and `completed` also in failure test

**Comment by iamareebjamal** [comment]
That'd be repetition of test logic. Just use pending in one test and placed in another and that'd be enough IMO

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/all/integration/api/helpers/order/test_edit_order.py  2
- app/api/orders.py  2
         

Clones added
============
- tests/all/integration/api/helpers/order/test_edit_order.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712&bid=18997949)

**Comment by iamareebjamal** [comment]
Add success case as well

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/all/integration/api/helpers/order/test_edit_order.py  3
- app/api/orders.py  2
         

Clones added
============
- tests/all/integration/api/helpers/order/test_edit_order.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5803712&bid=18997949)

**Pull request merged**